esphome/automation.py

```diff
@@ -303,6 +306,38 @@ async def component_update_action_to_code(config, action_id, template_arg, args)
     return cg.new_Pvariable(action_id, template_arg, comp)
 
 
+@register_action(
+    "component.suspend",
+    SuspendComponentAction,
+    maybe_simple_id(
+        {
+            cv.Required(CONF_ID): cv.use_id(cg.PollingComponent),
+        }
+    ),
+)
+async def component_suspend_action_to_code(config, action_id, template_arg, args):
+    comp = await cg.get_variable(config[CONF_ID])
+    return cg.new_Pvariable(action_id, template_arg, comp)
+
+
+@register_action(
+    "component.resume",
+    ResumeComponentAction,
+    maybe_simple_id(
+        {
+            cv.Required(CONF_ID): cv.use_id(cg.PollingComponent),
+            cv.Optional(CONF_UPDATE_INTERVAL): cv.positive_time_period_milliseconds,
```

Comment:
In the cpp class you define this as `templatable`, but the config side does not allow the use of a lambda to set.
```suggestion
            cv.Optional(CONF_UPDATE_INTERVAL): cv.templatable(cv.positive_time_period_milliseconds),
```

This also needs some changes in function to match

Reply:
Updated, along side the test it runs as (i) expected it to work